tfx/components/example_gen/base_example_gen_executor.py

```diff
@@ -33,13 +34,16 @@
 
 
 def _GeneratePartitionKey(record: Union[tf.train.Example,
-                                        tf.train.SequenceExample, bytes],
+                                        tf.train.SequenceExample, bytes,
+                                        Dict[str, Any]],
                           split_config: example_gen_pb2.SplitConfig) -> bytes:
   """Generates key for partition."""
 
   if not split_config.HasField('partition_feature_name'):
     if isinstance(record, bytes):
       return record
+    if isinstance(record, dict):
+      return pa.serialize(record).to_buffer().to_pybytes()
```

Comment:
Hi, Ihor just noticed the pa.serialize is deprecated, would pickle.dumps work here? or use the json_utils we have?

what's in the Any of Dict[str, Any]?

https://arrow.apache.org/docs/python/generated/pyarrow.serialize.html

Reply:
@1025KB done in https://github.com/tensorflow/tfx/pull/4937/commits/eaf408ed057e6ca4eea23b974648d4cd8bae3955